src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -1743,14 +1746,10 @@ internal void WriteInt(int v, TdsParserStateObject stateObj)
 
         internal static void WriteInt(Span<byte> buffer, int value)
         {
-#if NETCOREAPP
-            BitConverter.TryWriteBytes(buffer, value);
```

Comment:
Why remove the netcore special case rather than using BinaryPrimitives.WriteInt32LittleEndian in place of BitConverter.TryWriteBytes?

Reply:
The idea is to use the most performant code for methods such as WriteInt because some of the code you modified can run in a hot path/loop and affect performance. I'm not sure about WriteInt and the difference here is probably negligible but the places where you call GetBytes twice in the same row will certainly affect performance in some use cases, for example.

Reply:
If it's possible to give both big and little endian paths good performance and switch between inline them using BitConverter.IsLittleEndian then that is the preferred method. The jit will eliminate any unused code paths.

If getting equal (at best) or adequate (at worst) perf for big endian systems requires a preprocessor definition and that change is done in multiple places, then abstracting that to a another method is preferred.

The overall goal is to get equal performance wherever possible. 
We should not degrade performance for existing targets to enable new targets unless it is unavoidable.

Reply:
done